stackslib/src/chainstate/stacks/block.rs

```diff
@@ -804,7 +862,66 @@ impl StacksMessageCodec for StacksMicroblock {
         let header: StacksMicroblockHeader = read_next(fd)?;
         let txs: Vec<StacksTransaction> = {
             let mut bound_read = BoundReader::from_reader(fd, MAX_MESSAGE_LEN as u64);
-            read_next(&mut bound_read)
+            // The latest epoch where StacksMicroblock exist is Epoch25
+            read_next_at_most_with_epoch(&mut bound_read, u32::MAX, StacksEpochId::Epoch25)
```

Comment:
Should this use the `epoch_id` of the current scope?
```suggestion
            read_next_at_most_with_epoch(&mut bound_read, u32::MAX, epoch_id)
```

Reply:
I actually don't think this call is necessary at all.  You are already going to do epoch-specific checks in this function to determine if the transactions are _semantically_ valid within the epoch, so you should just decode them with `read_next_at_most()` and then deal with them below.